packages/transformer/src/IModelTransformer.ts

```diff
@@ -2890,19 +2914,67 @@ export class IModelTransformer extends IModelExportHandler {
         )
           continue;
         await this.processDeletedOp(
-          change,
+          change.ECInstanceId,
+          change.$meta?.classFullName ?? "",
+          change.FederationGuid,
           elemIdToScopeEsa,
           relationshipECClassIds.has(ecClassId ?? ""),
+          change.SourceECInstanceId,
+          change.TargetECInstanceId,
           alreadyImportedElementInserts,
           alreadyImportedModelInserts
         );
+        deleteIdsProcessed.add(change.ECInstanceId);
       }
 
       csReader.close();
     }
+
+    // This loop is to process all custom deleteIds. Unclear if the special logic is still necessary for relationships or not (TODO!!). For all other entities, we assume that the element is still present in the sourceDb because it is not
+    // a real delete and instead a simulated delete to update filtering criteria between source and target. Since the element is still present, we do not need to call processDeletedOp to find the corresponding targetId.
+    // We can instead rely on `forEachTrackedElement` at the top of processChangesets to find the corresponding targetId.
+    // Note this also assumes we don't need to handle entity recreation for these custom deletes. I.e. a caller of API would not be able to add a custom delete for an entity that was recreated.
+    // a delete followed by an insert.
+    // ASSUME: If a changeset has a deleteId then custom change will never reference it. Is this still true if it was re-inserted? (TODO!!)
```

Comment:
Im still unsure about entity recreation.

Basically a delete followed by an insert of the same entity (They would have different IDs, but may be considered the same if they had the same federationGuid).

What would you guys do with that with regards to customChanges?

Would it be possible for us to have that in our changesets (entity recreation) and then also have you decide that the insert (after the delete) should not make it to the target?

I think in theory the target would have the entity in it already... so you could potentially have a scenario where the entity was recreated in the source, but you know you also want to simulate a delete so that we get rid of it in the target. That scenario seems difficult to me.

(im not sure if im making sense)

Perhaps one option is to not let customChanges coexist with actual changes? Not sure how annoying that would be for you because I could see it meaning you have to run two transforms, one to get the changesets processed adn then one to update filter criteria. 

Theres probably another option we could come up with, but a little unsure how you guys plan to use customChanges in this specific scenario.

Reply:
**"Would it be possible for us to have that in our changesets (entity recreation) and then also have you decide that the insert (after the delete) should not make it to the target?"**
Yes, this is very possible. I think we should also add test case for such scenario. When filtering parameters change, it is possible that all elements that were previously added to target, now need to be removed. If e.g. in 1 transformation elements of "Category1" were exported, and now filtering parameters changed to export only "category 2" elements. 


**"Perhaps one option is to not let customChanges coexist with actual changes? Not sure how annoying that would be for you because I could see it meaning you have to run two transforms, one to get the changesets processed and then one to update filter criteria."** 

Not sure I fully understand how it would work, we would need to discuss this possibility a bit more.

Considerations I have about running 2 separate transformations, are:
- there is no support to run 2 transformation at the moment
- I don't plan to store/use old filter criteria, so first transformation would run with new new filter criteria. and then second transformation would do additional inserts / deletes as custom changes.  Would need to make more testing to see how that would work.   
- Support to run transformation when there are no changes in source would be needed for second transformation (as there would be only custom changes). But this will be needed anyway, so probably not a big blocker.